### PR TITLE
[#20] Manuals for 0.1.0

### PR DESCRIPTION
**Pull request by GuChad369**
Initial version for the 0.1.0 manual:

- Follow the core project manual to update the current manual.
- Add two new function tests.

### CODE REVIEW

File: doc/manual/02-installation.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+``` sh
+git gcc cmake gmake postgresql13-server postgresql13-client
+```
+
+### Build
+
+#### Release build
+
+The following commands will install [**pgmoneta_ext**][pgmoneta_ext] in the `/usr/local` hierarchy.
```
Comment (jesperpedersen) [inline doc/manual/02-installation.md]: It shouldn't be `/usr/local`, right ?

File: doc/manual/02-installation.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+```sh
+git clone https://github.com/pgmoneta/pgmoneta_ext
+cd pgmoneta_ext
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=/usr/local ..
```
Comment (jesperpedersen) [inline doc/manual/02-installation.md]: Same

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: doc/manual/02-installation.md (hunk 1 above)
Comment (GuChad369) [reply doc/manual/02-installation.md]: It doesn't matter where you install the extension. Wherever you install it, the extension will always load into PostgreSQL to make it work.

Comment (GuChad369) [reply doc/manual/02-installation.md]: In practice, we don't need to specify the download location. Here, we are just simulating pgmoneta to ensure they install together.

Comment (jesperpedersen) [reply doc/manual/02-installation.md]: Well, the extension should go into the PostgreSQL installation that is what all the cmake/FindPsqlDevel.cmake stuff is for...

Comment (GuChad369) [reply doc/manual/02-installation.md]: In our `CMake` file, we already use `pg_config` to determine important installation paths like `--sharedir`, `--includedir`, `--pkglibdir`, etc. These paths correspond to where PostgreSQL expects its extensions to be installed.

Here, we can remove the `-DCMAKE_INSTALL_PREFIX` option and not specify the installation location.

Comment (jesperpedersen) [reply doc/manual/02-installation.md]: Precisely

File: doc/manual/97-acknowledgement.md [hunk 3]
```diff
@@ -4,13 +4,21 @@
 
 ## Authors
 
-pgmoneta_ext was created by the following authors:
+[**pgmoneta**][pgmoneta] was created by the following authors:
```
Comment (jesperpedersen) [inline doc/manual/97-acknowledgement.md]: pgmoneta_ext

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
You have 2.6 Verify success that can be moved to 3 Quick start.

And, in 3 Quick start you can remove all the compile commands

**Comment by jesperpedersen** [comment]
You have the `postgresql.conf` requirement in chapter 2 instead of chapter 3.

And, you don't need a section in chapter 3 - everything can be in one

File: doc/manual/02-installation.md [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+```
+
+``` sh
+dnf install -y pgmoneta_ext
+```
+
+You need to add the `pgmoneta_ext` library for PostgreSQL in `postgrersql.conf` as well:
+
+```ini
+shared_preload_libraries = 'pgmoneta_ext'
```
Comment (jesperpedersen) [inline doc/manual/02-installation.md]: They are still here

File: doc/manual/user-01-quickstart.md [hunk 5]
```diff
@@ -2,53 +2,35 @@
 
 # Quick start
 
-## Compiling the source
+## Verify success
```
Comment (jesperpedersen) [inline doc/manual/user-01-quickstart.md]: Still here

File: doc/manual/user-01-quickstart.md [hunk 6]
```diff
@@ -2,53 +2,35 @@
...
 
-We recommend using `Fedora` to test and run pgmoneta_ext.
+You can use the `postgres` role to test.
```
Comment (jesperpedersen) [inline doc/manual/user-01-quickstart.md:13]: You are missing the `postgresql.conf` part

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by GuChad369** [comment]
Could you specify the file you're referring to, instead of using chapter names? I'm a bit confused right now.

**Comment by GuChad369** [comment]
> You have 2.6 Verify success that can be moved to 3 Quick start.
> 
> And, in 3 Quick start you can remove all the compile commands

For this part, have I achieved the desired output correctly?

> You have the `postgresql.conf` requirement in chapter 2 instead of chapter 3.
> 
> And, you don't need a section in chapter 3 - everything can be in one

For this part, I thought you wanted to move all **Configuration** in the file `dev-01-config.md` into the file `02-installation.md` section.

**Comment by jesperpedersen** [comment]
As above,

* doc/manual/02-installation.md - has `postgresql.conf` info
* doc/manual/user-01-quickstart.md - doesn't have `postgresql.conf` info for `shared_...` and has a section

**Comment by jesperpedersen** [comment]
Put on your user hat - Chapter 1. What is the extension ? Chapter 2. How do I install it, and maybe compile it ? Chapter 3. How do I configure it, and test it ?

Put on your developer hat - what does a developer who would like to help out need to know,  so the developer guide can be more "hard-core"

File: doc/manual/dev-01-config.md [hunk 7]
```diff
@@ -193,7 +109,7 @@ The following SQL command can revoke the superuser privileges:
 ALTER USER repl WITH NOSUPERUSER;
 ```
 
-### Check
+### 10. Check
```
Comment (jesperpedersen) [inline doc/manual/dev-01-config.md:112]: You need the `postgresql.conf` change for `shared_...` here as well

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by GuChad369** [comment]
I moved all tests under the developer manual. 

For users, they only need the functionality of the extension; they don't need to run tests. If some functions don't work, they can manually test them according to the `user-02-functions.md` file. 

For developers, who may need to modify the code, the test suite is designed specifically for them.

And the container test suite will always work successfully, so it is not necessary for users to run it.

**Comment by jesperpedersen** [comment]
There are some issues with the links in https://github.com/pgmoneta/pgmoneta_ext/blob/main/doc/manual/dev-02-git.md#git-guide in initial part

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Pull request merged**